src/Flame/Types.purs

```diff
@@ -77,13 +79,30 @@ isNonEventData = DA.filter case _ of
                         Property _ _ -> true
                         _ -> false
 
+
+type HookFn1 = VNode -> Effect Unit
+type HookFn2 = VNode -> VNode -> Effect Unit
+type HookFnRemove = VNode -> Effect Unit -> Effect Unit
```

Comment:
Lemme know if this isn't feasible, but If the hooks are to be only used in foreign calls, why not 
```
type HookFn1 = EffectFn1 VNode Unit
type HookFn2 = EffectFn2 VNode VNode Unit
type HookFnRemove = EffecFn2 VNode (Effect Unit) Unit  
```
I think you could then use regular `Effect.Uncurried` functions instead of Hook.js, similar to how node events are handled.

Reply:
Sure, as we use foreign hooks, that will make them more natural to JS and even let us get rid of foreign code in Flame.Renderer.Hook. Will do.